src/qml/editorwidgets/ValueMap.qml

```diff
@@ -26,38 +26,152 @@ EditorWidgetBase {
   height: childrenRect.height
   enabled: isEnabled
 
-  RowLayout {
-    anchors { left: parent.left; right: parent.right }
+  states: [
+    // showing QfToggleButton without search
+    State {
+      name: "toggleButtonsView"
+      PropertyChanges {
+        target: toggleButtons
+        visible: true
+      }
+      PropertyChanges {
+        target: comboBox
+        visible: false
+      }
+      PropertyChanges {
+        target: searchButton
+        visible: false
+      }
+    },
+    // showing ComboBox with search option
+    State {
+      name: "comboBoxItemView"
+      PropertyChanges {
+        target: toggleButtons
+        visible: false
+      }
+      PropertyChanges {
+        target: comboBox
+        visible: true
+      }
+      PropertyChanges {
+        target: searchButton
+        visible: searchButton.enabled
+      }
+    }
+  ]
 
-    ComboBox {
-      id: comboBox
+  state: currentItemCount >= minimumItemCount ? "comboBoxItemView" : "toggleButtonsView"
+
+  // Using the search and comboBox when there are less than X items in the dropdown proves to be poor UI on normally
+  // sized and oriented phones. Some empirical tests proved 6 to be a good number for now.
+  readonly property int minimumItemCount: 6
+  property real currentItemCount: comboBox.count
+  property bool testSwitch: true
+
+  ValueMapModel {
+    id: listModel
+    onMapChanged: {
+      comboBox.currentIndex = keyToIndex(valueMap.currentKeyValue)
+    }
+  }
 
+  RowLayout {
+    anchors {
+      left: parent.left
+      right: parent.right
+    }
+
+    Item {
+      id: toggleButtons
       Layout.fillWidth: true
+      Layout.minimumHeight: flow.height
+      opacity: enabled ? 1 : .4
 
-      font: Theme.defaultFont
+      property real selectedIndex: comboBox.currentIndex
+      property string currentSelectedKey: ""
+      property string currentSelectedValue: ""
 
-      popup.font: Theme.defaultFont
-      popup.topMargin: mainWindow.sceneTopMargin
-      popup.bottomMargin: mainWindow.sceneTopMargin
+      Flow {
+        id: flow
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.top: parent.top
+        spacing: 8
 
-      currentIndex: model.keyToIndex(value)
+        Repeater {
+          id: repeater
+          model: config['map']
 
-      model: ValueMapModel {
-        id: listModel
+          delegate: Rectangle {
+            id: item
+            width: innerText.width + 16
```

Comment:
We need to insure the rectangle will never be wider than the parent container (i.e. flow). So here I'd suggest doing something like `Math.min( flow.width - 16, innerText.width + 16 )`.